### DISCUSSION TITLE
Compile-time / lazy loaded guards

### DISCUSSION THREAD

**Discussion by SteelAlloy** [async-graphql/async-graphql#1253] [Q&A] [answered]
Would it be possible to have compile-time guards ? Or maybe initialized once (the first time the guard is called). In some case it can be expensive to init the guard and takes time to do it on each request.
```rust
    #[graphql(guard = "SomeMemoryExpensiveGuard::new(16)")]
    async fn name(&self) -> &String {
        &self.name
    }
```

I could have done it with `lazy_static` but it's not very convenient to have one constant by field. 

Thanks a lot !

**Comment by Yatekii** [comment]
You can use https://docs.rs/once_cell/latest/once_cell/ :)

**Reply by SteelAlloy** [reply]
Hi, thanks for the help.
I tried for a few hours to tinker with this lib but couldn't come up with something that worked in my use case.
Could you provide an example ?

**Reply by tgross35** [reply, marked as answer]
@oganexon you can use `once_cell::sync::Lazy::new(|| some_slow_function())` and put it in a static. That will call the slow function the first time the `Lazy` is used.

They have some examples in their docs https://docs.rs/once_cell/latest/once_cell/sync/struct.Lazy.html#impl-Deref-for-Lazy%3CT,+F%3E, not async_graphql specific but shouldn't be hard to adapt
